Library/Homebrew/formula.rb

```diff
@@ -927,6 +927,40 @@ def unlock
   end
 
   # @private
+  def outdated_versions
+    all_versions = []
+    older_or_same_tap_versions = []
+
+    if oldname && !rack.exist? && (dir = HOMEBREW_CELLAR/oldname).directory? &&
+      !dir.subdirs.empty? && tap == Tab.for_keg(dir.subdirs.first).tap
+      raise Migrator::MigrationNeededError.new(f)
+    end
+
+    rack.subdirs.each do |keg_dir|
+      keg = Keg.new keg_dir
+      version = keg.version
+      all_versions << version
+      older_version = pkg_version <= version
+
+      tap = Tab.for_keg(keg).tap
+      if tap.nil? || tap == tap || older_version
+        older_or_same_tap_versions << version
+      end
+    end
+
+    return [] unless older_or_same_tap_versions.all? { |v| pkg_version > v }
+
+    all_versions
```

Comment:
`Formula#outdated_versions` will return `all_versions`?

Maybe we should make them two methods, i.e. `Formula#alll_versions` and `Formula#outdated_versions`.